planner/core/plan_cache.go

```diff
@@ -662,6 +662,10 @@ func buildRangeForTableScan(sctx sessionctx.Context, ts *PhysicalTableScan) (err
 }
 
 func buildRangeForIndexScan(sctx sessionctx.Context, is *PhysicalIndexScan) (err error) {
+	if len(is.IdxCols) == 0 { // return full range when PhysicalIndexScan has no IdxCols
+		is.Ranges = ranger.FullRange()
```

Comment:
```
if IsFullRagne(is.Ranges) { // the original range is already a full-range.
    return nil
} else {
    return error("unexpected")
}
```

Reply:
@qw4990 Do you mean that we should assert `IsFullRagne(is.Ranges)` must stand when `len(is.IdxCols) == 0`?

Reply:
Yes, I'm not sure whether `len(is.IdxCols)==0` can guarantee this is a `IndexFullScan`, so check `IsFullRagne(is.Ranges)` for safety.

Reply:
got it